### PR TITLE
[Monitor] Diagnostic settings commands

### PR DESCRIPTION
**Pull request by vcanaa**
## Description

Renaming the ServiceDiagnosticSettings operation to DiagnosticSettings operation and adding more functionality.

- The new command will be backward compatible with the existing api /diagnosticSettings/service and will also works with /diagnosticSettings/{name}, where a generic name can be passed in.

- Renaming ServiceBusRuleId to EventHubAuthorizationRuleId.

- Accepting EventHubName to be defined

- Adding List operation to diagnostic settings.

- Adding operations to list diagnostic settings categories.

https://github.com/Azure/azure-rest-api-specs/pull/1641

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@vcanaa  please open this PR once you have sorted out Spec API referenced.
Seems your rest spec was not merged.
Reopen this PR with the update spec PR link

**Comment by vcanaa** [comment]
I fixed the reference to the spec. There is no link for me to reopen it. 

Please reopen this PR.

**Comment by shahabhijeet** [comment]
@vcanna please co-ordinate with @gucalder and have a single nuget package published.
We do not want to publish multiple packages within days.

**Comment by vcanaa** [comment]
@shahabhijeet - We will not publish multiple packages.

This was marked as Changes Requested, but I don't understand what changes are necessary. What is blocking this to be merged right now?

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <Description>Microsoft Azure Monitor Library</Description>
     <AssemblyName>Microsoft.Azure.Management.Monitor</AssemblyName>
-    <Version>0.16.1-preview</Version>
+    <Version>0.17.0-preview</Version>
```
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: can you update the release notes
Add < PackageReleaseNotes > and add Release notes.
I would recommend to use < ![CDATA and then format your Release notes

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj [hunk 2]
```diff
@@ -3,9 +3,10 @@
...
     <PackageId>Microsoft.Azure.Management.Monitor</PackageId>
     <PackageTags>Management.Monitor;Management.Monitoring;metrics;alerts;autoscale;activityLogs;events;operations;logs</PackageTags>
+    <PackageReleaseNotes>See https://github.com/Azure/azure-sdk-for-net/tree/psSdkJson6/src/SDKs/Monitor/changelog.md for release notes.</PackageReleaseNotes>
```
Comment (shahabhijeet) [inline src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: This URL will be of no use once psSdkJson6 branch is deleted
Rather create an aka.ms link for your changelog.md
Will allow it for now, but consider the above option for future releases

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj (hunk 2 above)
Comment (vcanaa) [reply src/SDKs/Monitor/Management.Monitor/Microsoft.Azure.Management.Monitor.csproj]: Indeed, thanks for catching that. Will update to:
https://aka.ms/azure-sdk-monitor

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**